datasets/codexglue_concode/codexglue_concode.py

```diff
@@ -0,0 +1,92 @@
+"""concode - code2text pairs dataset"""
+import datasets
+import json
+import logging
+
+_CITATION = """\
+@article{iyer2018mapping,
+  title={Mapping language to code in programmatic context},
+  author={Iyer, Srinivasan and Konstas, Ioannis and Cheung, Alvin and Zettlemoyer, Luke},
+  journal={arXiv preprint arXiv:1808.09588},
+  year=2018,
+}
+"""
+_URL = "https://raw.githubusercontent.com/microsoft/CodeXGLUE/main/Text-Code/text-to-code/dataset/concode/"
+_URLS = {
+    "train": _URL + "train.json",
+    "dev": _URL + "dev.json",
+    "test": _URL + "test.json",
+}
+_DESCRIPTION = """Mapping Language to Code in a Programmatic Context"""
+class CodeXGlue_ConcodeConfig(datasets.BuilderConfig):
+    """BuilderConfig for Concode Dataset."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for Concode.
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(CodeXGlue_ConcodeConfig, self).__init__(**kwargs)
+
+class Concode(datasets.GeneratorBasedBuilder):
+    """ConCode: code generation dataset for java NL queries(pre-processed)"""
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        CodeXGlue_ConcodeConfig(
+            name="plain_text",
+            version=VERSION,
+            description="Plain text",
+        ),
+    ]
+    def _info(self):
+            return datasets.DatasetInfo(
+                description=_DESCRIPTION,
+                features=datasets.Features(
+                    {
+                        "nl": datasets.Value("string"),
```

Comment:
what's nl exactly ?
If it is the language name, maybe we should name it `language` and use ClassLabel feature type ?